### PR TITLE
Allow users to be destroyed

### PR DESCRIPTION
**Pull request by allenwq**
@lowjoel Seems rails is using inheritance to implement callbacks: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/callbacks.rb#L292

There is no status indicating the record is being destroyed, so `dependent: :delete_all` might be a good solution, otherwise we should not use callbacks.

### CODE REVIEW

**Comment by lowjoel** [comment]
There is another thing you can do, since it is implemented using callbacks, you can define an AR::Base extension that hooks around_destroy that sets `@destroying` on entrance and has a `destroying?` method to query.

**Comment by lowjoel** [comment]
Does #434 depend on #439?

**Comment by allenwq** [comment]
@lowjoel yep, #434 was depend on this

**Comment by lowjoel** [comment]
I'm concerned delete_all might not work because we will be using callbacks to handle versioning.

**Comment by allenwq** [comment]
@lowjoel Thanks, and I've updated.

File: lib/extensions/destroy_callbacks/active_record/base.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Extensions::DestroyCallbacks::ActiveRecord::Base
+  extend ActiveSupport::Concern
+  included do
+    around_destroy :update_status
+  end
+
+  # @return [boolean] True if the record is being destroyed.
+  def destroying?
```
Comment (lowjoel) [inline lib/extensions/destroy_callbacks/active_record/base.rb]: @allenwq Does Rubocop complain?

Comment (allenwq) [inline lib/extensions/destroy_callbacks/active_record/base.rb]:
```
> rubocop lib/extensions/destroy_callbacks/active_record/base.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected

```

Nope

**Pull request merged**